Userland/Utilities/lsirq.cpp

```diff
@@ -10,27 +10,51 @@
 #include <LibCore/System.h>
 #include <LibMain/Main.h>
 
+static ErrorOr<size_t> system_cpu_count()
+{
+    auto file = TRY(Core::File::open("/sys/kernel/cpuinfo", Core::OpenMode::ReadOnly));
+    auto buffer = file->read_all();
+    auto json = TRY(JsonValue::from_string({ buffer }));
+    return json.as_array().size();
+}
+
 ErrorOr<int> serenity_main(Main::Arguments)
 {
     TRY(Core::System::pledge("stdio rpath"));
+    TRY(Core::System::unveil("/sys/kernel/cpuinfo", "r"));
     TRY(Core::System::unveil("/sys/kernel/interrupts", "r"));
     TRY(Core::System::unveil(nullptr, nullptr));
 
+    auto cpu_count = TRY(system_cpu_count());
+
     auto proc_interrupts = TRY(Core::File::open("/sys/kernel/interrupts", Core::OpenMode::ReadOnly));
 
     TRY(Core::System::pledge("stdio"));
 
-    outln("      CPU0");
+    {
+        StringBuilder builder;
+        builder.append("      "sv);
+        for (size_t i = 0; i < cpu_count; ++i) {
+            builder.appendff("{:>10}", String::formatted("CPU{}", i));
+        }
+        outln("{}", builder.to_string());
```

Comment:
Wouldn't calls to `out` suffice here,
I see no need to use a builder in this case

Reply:
Sure!